core/src/main/java/org/infinispan/util/concurrent/locks/impl/DefaultPendingLockManager.java

```diff
@@ -77,14 +83,14 @@ public PendingLockPromise checkPendingTransactionsForKey(TxInvocationContext<?>
          return createAndStore(getTransactionWithLockedKey(txTopologyId, key, globalTransaction),
                                globalTransaction, time, unit);
       }
-      return PendingLockPromise.NO_OP;
+      return createAndStoreDefault(globalTransaction, time, unit);
```

Comment:
I don't quite understand why you need to put these promises in the map, could you add a comment explaining that?